modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/EndpointHelper.java

```diff
@@ -145,6 +146,10 @@ private Object _getRelatedObjectValue(
 			return objectEntry.getPropertyValue(property.getSourceFieldName());
```

Comment:
I think the if clause validating the null value of the objectEntry should be the first one, as here you are acessing the objectEntry

Reply:
Make sense! Closing the Brian's PR and resend